jaeger-core/src/main/java/com/uber/jaeger/Span.java

```diff
@@ -267,19 +267,46 @@ private Span setTagAsObject(String key, Object value) {
   }
 
   @Override
+  public Span log(Map<String, ?> fields) {
+    return log(tracer.clock().currentTimeMicros(), fields);
+  }
+
+  @Override
+  public Span log(long timestampMicroseconds, Map<String, ?> fields) {
+    synchronized (this) {
+      if (context.isSampled()) {
+        if (logs == null) {
+          this.logs = new ArrayList<>();
+        }
+        logs.add(new LogData(timestampMicroseconds, "", fields));
```

Comment:
Self-inflicted inefficiency here, you reuse (String, Object) signature and then are forced to check for instanceof Map. Not to mention that there may be a payload that's a map.

I suggest simply having another constructor for LogData that stores `Map<String, ?>` fields proper.